CMU-Robotics-Club/RoboBuggy2
# Changes made during mock rolls, debugging sessions that need to be preserved

## What type of PR is this? (check all applicable)


## Description
This branch contains utilities for mock rolls and debugging, fixes for minor bugs discovered during mock rolls, as well as adjustments to auton parameters.

Fixed minor bugs in launch files.
Added launch files and nodes for debugging auton overtake, filter separation and firmware steering.
Added path files for the cut and UC to purnell footpath. 
Set debug to true in INS parameters.
Adjusted MPC path index search range.
Added exception in MPC and Stanley when the buggy reaches the end of the path.
Changed topic names to be consistent with firmware


## Related Tickets & Documents



- Related Issue #
- Closes #

## QA Instructions, Screenshots, Recordings

This branch has been tested in mock rolls up to the commit where the current main branch was merged. Changes made during and after the merge are not tested in mock rolls.

Tests performed and passed pre-merging:

1. run debug_filter.launch to check if the topic "/gnss_odom_distance" reports separation between the front antenna and the filter estimate. (requires going outside with SC)
2. run debug_steer.launch to check if the topic "/SC/buggy/input/steering" has an oscillating pattern (plot on Foxglove).
3. run main.launch with NAND_exist = false and roll the buggy. 
5. run ghost_auton.launch with path=UC_to_purnell.json, start the buggy in front of UC courtyard and check if the buggy avoids the ghost NAND.

In order to full test this PR, these tests should be repeated during a Mock Roll.
